roles/etcd_prometheus_exporter_cert/tasks/gen_cert.yml

```diff
@@ -0,0 +1,85 @@
+- name: 'make assertions'
+  assert:
+    that:
+     - groups.etcd is defined
+
+- name: 'create private key on master'
+  command: openssl genrsa -out {{ etcd_exporter_key }} 4096
+  args:
+    creates: '{{ etcd_exporter_key }}'
+  run_once: True
+  register: gen_etcd_exporter_key
+
+- name: 'test if etcd_exporter cert already exists'
+  stat:
+    path: '{{ etcd_exporter_cert }}'
+  register: etcd_exporter_cert_file
+
+- name: 'generate csr'
+  command: >-
+    openssl req -new
+    -subj '/CN=etcd-exporter'
+    -key {{ etcd_exporter_key }}
+  run_once: True
+  register: gen_etcd_exporter_csr
+  when: >-
+    gen_etcd_exporter_key is changed or not etcd_exporter_cert_file.stat.exists
+
+- debug:
+    var: gen_etcd_exporter_csr
+  when: debug|bool
+
+- name: 'copy csr over a server capable of signing etcd cert'
+  copy:
+    dest: '/tmp/ansible.{{ lookup("password", "/dev/null chars=ascii_letters length=8") }}'
+    content: '{{ gen_etcd_exporter_csr.stdout }}'
+  run_once: True
+  delegate_to: '{{ groups.etcd|first }}'
+  when: gen_etcd_exporter_csr is changed
+  register: csr_on_etd_node
+
+- debug:
+    var: csr_on_etd_node
+  when: debug|bool
+
+- name: 'generate certificate'
+  command: >-
+    openssl x509 -req -days 3650
+    -in {{ csr_on_etd_node.dest }}
+    -CAkey {{ etcd_cert_dir }}/ca-key.pem
+    -CA {{ etcd_cert_dir }}/ca.pem
+    -CAcreateserial
+  run_once: True
+  delegate_to: '{{ groups.etcd|first }}'
+  when: gen_etcd_exporter_csr is changed
+  register: gen_etcd_exporter_cert
+
+- debug:
+    var: gen_etcd_exporter_cert
+  when: debug|bool
+
+- name: 'copy cert on master node'
+  copy:
+    dest: '{{ etcd_exporter_cert }}'
+    content: '{{ gen_etcd_exporter_cert.stdout }}'
+  run_once: True
+  when: gen_etcd_exporter_cert is changed
+
+- name: 'download cert from signing server '
+  slurp:
+    path: '{{ etcd_exporter_cert }}'
+  run_once: True
+  register: etcd_exporter_cert_content
+
+- name: 'copy cert on master node'
+  copy:
+    dest: '{{ etcd_exporter_cert }}'
+    content: '{{ etcd_exporter_cert_content.content|b64decode }}'
```

Comment:
Why `b64encode`? I know this is required when putting the cert etc. in a YAML manifest when creating the `Secret`, but why encode in a PEM-file which one doesn't expect to be encoded in any way? This makes it more difficult to use the cert with `curl` or similar to troubleshoot things.